kratos/utilities/qef_utility.h

```diff
@@ -0,0 +1,157 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortes
+//
+//
+
+#if !defined(KRATOS_QEF)
+#define  KRATOS_QEF
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/ublas_interface.h"
+#include "includes/geometrical_object.h"
+#include "includes/node.h"
+#include "geometries/geometry.h"
+#include "geometries/hexahedra_3d_8.h"
+#include "geometries/triangle_3d_3.h"
+#include "intersection_utilities.h"
+#include "../external_libraries/a_matrix/include/matrix.h"
+
+namespace Kratos
+{ 
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class QEF (quadratic error function)
+ * @ingroup KratosCore
+ * @brief Utilities to compute the minimum error point in a 3D voxel intersected by a triangle mesh
+ * @author Ariadna Cortes
+ */
+class KRATOS_API(KRATOS_CORE) QEF
```

Comment:
Rather than creating a new file to implement only one function, I'd add this to the `VoxelUtitlities` I suggested in your other PR.

Reply:
I think it would be better to leave the QEF utility separetely from Voxel, since it can be extended to any kind of geometry (in fact we will probably be using it soon for tetrahedra and other applications that are not strictly related to Voxels)

Reply:
OK. If this is the case it makes sense to me. Thanks for the clarification.